wordpress-mobile/WordPress-iOS
# Switch Fastlane to API calls

This PR wraps up the changes to trigger builds on CI via API calls on WPiOS. It's built on top of https://github.com/wordpress-mobile/WordPress-iOS/pull/15918 and should be merged after that one. 

The PR removes all the steps which push tags in `Fastlane` and replace them with API calls to the CI. 
CircleCI's `config.yml` is updated with filters which run the build lanes only on the proper calls. 


**To test**:
Testing this one is tricky. I think that it's quite easy to test the critical points, but a complete test of the flow would require many small tricks/adjustments. So: 

1. Verify CI is green. 
2. Double check that, in `Fastfile`, every call to `ios_tag_build()` and `ios_final_tag()` has been replaced with a call to `trigger_beta_build` and `trigger_release_build`.
3.  Create a new `release/test` branch out of this one and push it.
4. Run `bundle exec fastlane trigger_beta_build branch_to_build:"release/test"` (which is the same lane invoked by the other lanes in `Fastfile`) and verify that the build is triggered _and_ that the standard PR jobs are not triggered. Note that the build will likely fail because the built version already exists.
5. Remove the `release/test` branch and clean up.  

PR submission checklist:
